articles/golang-part-2-programming-basics/index.md

```diff
@@ -0,0 +1,304 @@
+
+![control flow](/engineering-education/golang-part-2-programming-basics/control-flow.jpg)
+
+The basic structure of conditionals:
+
+```
+if condition1 {
+    statements executed when condition1 is true
+} else if condition2 {
+    statements executed when condition2 is true and condition1 is false
+.
+.
+.
+} else {
+    statements executed when none of the conditions are true
+}
+```
+
+An "if" statement can be used without a corresponding else statement whereas an "else" statement must always follow an "if" statement. "else" statement cannot be used without a corresponding "if".
+
+```
+if condition {
+    statements
+}
+```
+
+An if-else block can be "nested" inside another if-else block as follows:
+
+```
+if condition1 {
+    statements
+
+    if condition2 {
+        statements
+    } else {
+        statements
+    }
+}
+```
+
+
+#### Loops
+[Loops](https://en.wikipedia.org/wiki/For_loop) are a set of instructions that are repeated until a certain condition is met. Loops can be used when the user wants to execute a block of statements multiple times.
+
+```go
+package main
+
+import (
+    "fmt"
+)
+
+func main() {
+
+    for i := 0; i <= 10; i++ {
+        fmt.Println(i)
+    }
+}
+```
+A for loop executes all the statements inside the curly brackets as many times as specified. In the above example, the for loop prints all the numbers from 0 to 10. The structure of a for loop is as follows:
+
+```
+for variable declaration; condition; operation {
+    statements
+}
+
+(while loop)
+for condition {
+    statements
+}
+
+```
+
+The loop only executes until the condition is satisfied. In the above example, we declare and initialize a variable called "i" to zero and execute the for loop until "i" reaches the value 10. The increment operator (++) is used to increment the value of "i" after every iteration of the loop.
+
+Go does not have a **while** loop. However, a for loop can act as a while loop in go:
+
+```go
+package main
+
+import (
+    "fmt"
+)
+
+func main() {
+
+    i := 0
+    for i <= 10 {
+        fmt.Println(i)
+        i++
+    }
+}
+```
+
+### Putting it all together
+Let us use the above concepts to write a program to add all the even numbers from 1 to 100 and print the result.
+
+```go
+package main
+
+import (
+    "fmt"
+)
+
+func main() {
+
+    var sum int = 0
+
+    for i := 1; i <= 100; i++ {
+
+        if i % 2 == 0 {
+            sum = sum + i
+        }
+    }
+    fmt.Println(sum)
+}
+```
+
+1. We define a variable called sum as an integer and initialize it to zero.
+
+2. We run a for loop to go through all the numbers from 1 to 100 (inclusive).
+
+3. In the for loop, we check to see if the current number is divisible by 2 (if i % 2 == 0). If it is divisible by 2, the statements inside the if block are executed.
```

Comment:
(if i % 2 == 0)

might look better using code formatting by wrapping with the ` symbol
`if i % 2 == 0`